attrd/commands.c

```diff
@@ -669,14 +669,41 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
 {
     bool changed = FALSE;
     attribute_value_t *v = NULL;
+    int dampen = 0;
 
+    const char *op = crm_element_value(xml, F_ATTRD_TASK);
     const char *attr = crm_element_value(xml, F_ATTRD_ATTRIBUTE);
     const char *value = crm_element_value(xml, F_ATTRD_VALUE);
+    const char *dvalue = crm_element_value(xml, F_ATTRD_DAMPEN);
 
     attribute_t *a = g_hash_table_lookup(attributes, attr);
 
     if(a == NULL) {
-        a = create_attribute(xml);
+        if (safe_str_eq(op, ATTRD_OP_UPDATE) || safe_str_eq(op, ATTRD_OP_UPDATE_VALUE_AND_DAMPEN)) {
+            a = create_attribute(xml);
+        }
+    } else {
+        if (safe_str_eq(op, ATTRD_OP_UPDATE_VALUE_AND_DAMPEN) || safe_str_eq(op, ATTRD_OP_UPDATE_ONLY_DAMPEN)) {
+            if (dvalue) {
+                dampen = crm_get_msec(dvalue); 
+                if (dampen >= 0) {
+                    if (a->timeout_ms != dampen) {
+                        mainloop_timer_stop(a->timer);
+                        mainloop_timer_del(a->timer);
+                        a->timeout_ms = dampen;
+                        if (dampen > 0) {
+                            a->timer = mainloop_timer_add(a->id, a->timeout_ms, FALSE, attribute_timer_cb, a);
+                            crm_info("Update attribute %s with delay %dms (%s)", a->id, dampen, dvalue);
+                        } else {
+                            a->timer = NULL;
+                            crm_info("Update attribute %s with not delay", a->id);
+                        }
+                        //if dampen is changed, attrd writes in a current value immediately.
+                        write_or_elect_attribute(a);
+                    }
+                } 
+            }
+        }
```

Comment:
When attrd throws processing up in the case of ATTRD_OP_UPDATE_ONLY_DAMPEN, in the case of next, an attribute is not updated.
It is not updated at all until an attribute is changed again.

1. I send the attribute together first in damepn=60.(attrd_updater -n test -v XXX -d 60)
1. While it does not pass for 60 seconds, I change dampen.(attrd_updater -Y -d 20)
1. As a result, because I changed dampen without waiting for the progress of first 60 seconds, the attribute continues being maintained until there is the change of the next attribute.(cib will not have test attribute until the change of the next value.)

What do you think?